docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,491 @@
+        -DENABLE_CUSOLVER_BACKEND=ON \
+        -DENABLE_CURAND_BACKEND=ON \
+        -DBUILD_FUNCTIONAL_TESTS=OFF
+
+`$ONEMKL_DIR` points at the oneMKL source directly.
+The x86 CPU (``MKLCPU``) and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled here.
+The backends for Nvidia GPUs must all be explicilty enabled.
+The tests are disabled, but the examples will still be built.
+
+Building oneMKL with support x86 CPU, Intel GPU, and AMD GPUs with tests disabled:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \
+        -DCMAKE_CXX_COMPILER=clang++ \
+        -DCMAKE_C_COMPILER=clang \
+        -DENABLE_MKLCPU_BACKEND=OFF \
+        -DENABLE_MKLGPU_BACKEND=OFF \
+        -DENABLE_ROCFFT_BACKEND=ON  \
+        -DENABLE_ROCBLAS_BACKEND=ON \
+        -DENABLE_ROCSOLVER_BACKEND=ON \
+        -DHIP_TARGETS=gfx90a \
+        -DBUILD_FUNCTIONAL_TESTS=OFF
+
+`$ONEMKL_DIR` points at the oneMKL source directly.
+The x86 CPU (``MKLCPU``) and Intel GPU (``MKLGPU``) backends are enabled by default, but are disabled here.
+The backends for AMD GPUs must all be explicilty enabled.
+The tests are disabled, but the examples will still be built.
+
+
+Build oneMKL for the DFT domain only with support for x86 CPU, Intel GPU, AMD GPU and Nvidia GPU with
+testing enabled:
+
+.. code-block:: bash
+
+    cmake $ONEMKL_DIR \ 
+        -DCMAKE_CXX_COMPILER=icpx \
+        -DCMAKE_C_COMPILER=icx \ 
+        -DENABLE_ROCFFT_BACKEND=ON \
+        -DENABLE_CUFFT_BACKEND=ON \
+        -DTARGET_DOMAINS=dft \
+        -DBUILD_EXAMPLES=OFF
+
+Note that this is not a supported configuration, and requires Codeplay's oneAPI for 
+`AMD <https://developer.codeplay.com/products/oneapi/amd/home/>`_ and 
+`Nvidia <https://developer.codeplay.com/products/oneapi/nvidia/home/>`_ GPU plugins.
+Like the above example, the MKLCPU and MKLGPU backends are enabled by default, with
+backends for Nvidia GPU and AMD GPU explicitly enabled.
+``-DTARGET_DOMAINS=dft`` causes only DFT backends to be built. If this was not set,
+the backend libraries to enable the use of BLAS, LAPACK and RNG with MKLGPU and MKLCPU
+would also be enabled.
+The build of examples is disabled.
+Since functional testing was not disabled, tests would be built.
+
+.. _project_cleanup:
+
+Project Cleanup
+###############
+
+Most use-cases involve building the project without the need to cleanup the
+build directory. However, if you wish to cleanup the build directory, you can
+delete the ``build`` folder and create a new one. If you wish to cleanup the
+build files but retain the build configuration, following commands will help
+you do so.
+
+.. code-block:: sh
+
+   # If you use "GNU/Unix Makefiles" for building,
+   make clean
+
+   # If you use "Ninja" for building
+   ninja -t clean
+
+
+.. _build_for_windows_dpcpp:
+
+Building for Windows
+####################
+
+The Windows build is similar to the Linux build, albeit that fewer backends are supported. For example:
```

Comment:
Can we list (or link to) exactly what is supported on windows?

Reply:
I've linked to the support table in [2b9a352](https://github.com/oneapi-src/oneMKL/pull/510/commits/2b9a352c5755b09fc8c395233f1f4393e54fdbe3).